tests/baselines/reference/thisInFunctionCall.types

```diff
@@ -84,7 +84,7 @@ class Test {
 
     /** @this {Test} */
     function (d) {
->function (d) {      console.log(d === this.data.length)    } : (this: Test, d: number) => void
+>function (d) {      console.log(d === this.data.length)    } : (d: number) => void

```

Comment:
Does this test need to be preserved somewhere with a .js extension so we can still cover this case, or do we have another test which does this correctly?

Reply:
I've added a test for the JavaScript case.